crates/glaredb/src/server.rs

```diff
@@ -8,40 +8,42 @@ use rpcsrv::flight_handler::{FlightServiceServer, FlightSessionHandler};
 use rpcsrv::{handler::RpcHandler, simple::SimpleHandler};
 use sqlexec::engine::{Engine, EngineStorageConfig};
 use std::collections::HashMap;
-use std::net::SocketAddr;
 use std::path::PathBuf;
 use std::sync::Arc;
 use std::{env, fs};
 use telemetry::{SegmentTracker, Tracker};
 use tokio::net::TcpListener;
 use tokio::signal;
 use tokio::sync::oneshot;
-use tonic::transport::server::Router;
+use tonic::transport::server::{Router, TcpIncoming};
 use tonic::transport::Server;
 use tracing::{debug, debug_span, error, info, Instrument};
 use uuid::Uuid;
 
-#[derive(Debug)]
-pub struct ServerConfig {
+pub struct PostgresProtocolConfig {
     /// Listener to use for pg handler.
-    pub pg_listener: TcpListener,
-
-    /// Address to use for the rpc handler. If not provided, an rpc handler will
-    /// not be started.
-    pub rpc_addr: Option<SocketAddr>,
+    pub listener: TcpListener,
+    pub handler: Arc<ProtocolHandler>,
```

Comment:
Document what this is. I don't know if `handler` is for the rpc stuff or something else.